app.rb

```diff
@@ -60,7 +61,12 @@ class AbodenB < Sinatra::Base
   end
 
   get "/listings" do
+    check_login
     @listings = Listing.all
     erb :listings
   end
+
+  def check_login
```

Comment:
Nice method, would it be a good idea to put this in a different helper file to have as little ruby logic as possible in the controller?